saleor/warehouse/availability.py

```diff
@@ -62,7 +63,14 @@ def check_stock_quantity_bulk(
         variant_stocks[stock.product_variant_id].append(stock)
 
     insufficient_stocks: List[InsufficientStockData] = []
-    for variant, quantity in zip(variants, quantities):
+    for variant, input_quantity in zip(variants, quantities):
+
+        quantity = input_quantity
+        if existing_lines:
+            for line_info in existing_lines:
```

Comment:
Instead of calling `for`  for each variant, wouldn't be good to create a dict of lines (before the first for)- `{"variant.pk": "line"}` ? This should simplify the logic and complexity.